File: src/main/kotlin/de/thm/mni/ii/classroom/event/TicketEvent.kt
package de.thm.mni.ii.classroom.event

import com.fasterxml.jackson.annotation.JsonIgnore
import de.thm.mni.ii.classroom.model.classroom.Ticket

data class TicketEvent(
    val ticket: Ticket,
    val ticketAction: TicketAction,
) : ClassroomEvent(TicketEvent::class.simpleName!!) {
    @JsonIgnore override fun getClassroomId(): String = this.ticket.classroomId
}

enum class TicketAction {
    CREATE,
    ASSIGN,
    CLOSE,
    EDIT
}

File: web-gui/src/app/rsocket/event/TicketEvent.ts
import {Ticket} from "../../model/Ticket";
import {ClassroomEvent} from "./ClassroomEvent";

export enum TicketAction {
  CREATE = "CREATE",
  ASSIGN = "ASSIGN",
  CLOSE = "CLOSE"
}

export class TicketEvent extends ClassroomEvent {
    ticket: Ticket
    ticketAction: TicketAction

    constructor(ticket: Ticket, action: TicketAction) {
      super("TicketEvent");
      this.ticket = ticket
      this.ticketAction = action
    }
}
